atomicapp/nulecule/base.py

```diff
@@ -548,4 +550,9 @@ def _get_artifact_paths_for_path(self, path):
                 if dir_child.startswith('.') or os.path.isdir(dir_child_path):
                     continue
                 artifact_paths.append(dir_child_path)
+        elif os.listdir(path) == []:
+            raise NuleculeException("Artifact directory %s is empty" % path)
+        else:
+            raise NuleculeException("Unable to find artifact %s" % path)
+
```

Comment:
Instead of putting our logic in the if/else statements maybe we should just check to see if artifact_paths is empty before we return and raise an exception there?